src/server/index.ts

```diff
@@ -12,33 +13,68 @@ import Storage from './storage';
 
 dotenv.config();
 const storage = new Storage();
+const actions = new Actions();
+
+const metrics: MetricsDefinition = {
+  data: {
+    balance: 0,
+    errors_rpc_timeout: 0,
+    errors_total: 0,
+    success_requests: 0,
+    total_requests: 0
+  },
+  meta: {
+    prefix: 'faucet'
+  }
+};
+
+/**
+ * Simplistic function to generate a prometheus metrics.
+ * TODO: Switch to prom-client
+ * @param name Name of the metric
+ * @param type Type
+ * @param value Value
+ * @param help Help test
+ * @param voidIfUndefined Whether we render it even if null/undefined
+ * @returns string
+ */
+function getMetrics (name: string, type: string, value: number | string | undefined, help = '', voidIfUndefined = false): string {
+  if (!value && voidIfUndefined) return '';
+  const metrics_name = `${metrics.meta.prefix}_${name}`;
+
+  let result = '';
+  if (help && help.length) result += `# HELP ${metrics_name} ${help}\n`;
+  result += `# TYPE ${metrics_name} ${type}\n`;
+  result += `${metrics_name} ${value || 0}\n`;
+
+  return result;
+}
 
 const app = express();
 app.use(bodyParser.json());
 
 checkEnvVariables(envVars);
 
-const port = getEnvVariable('PORT', envVars) as string;
+const port = getEnvVariable('PORT', envVars) as number;
 
 app.get('/health', (_, res) => {
   res.send('Faucet backend is healthy.');
 });
 
 // prometheus metrics
 app.get('/metrics', (_, res) => {
-  res.end(`# HELP errors_total The total amount of errors logged on the faucet backend
-# TYPE errors_total counter
-errors_total ${errorCounter.total()}
-# HELP errors_rpc_timeout The total amount of timeout errors between the faucet backend and the rpc node
-# TYPE errors_rpc_timeout counter
-errors_rpc_timeout ${errorCounter.getValue('rpcTimeout')}
-`
-  );
+  const errors_total = getMetrics('errors_total', 'counter', errorCounter.total(), 'The total amount of errors logged on the faucet backend');
+  const errors_rpc_timeout = getMetrics('errors_rpc_timeout', 'counter', errorCounter.getValue('rpcTimeout'), 'The total amount of timeout errors between the faucet backend and the rpc node');
+
+  const balance = getMetrics('balance', 'gauge', actions.getFaucetBalance(), 'Current balance of the faucet', true);
+
+  const total_requests = getMetrics('total_requests', 'gauge', metrics.data.total_requests, 'Total number of requests to the faucet');
+  const successful_requests = getMetrics('successful_requests', 'gauzge', metrics.data.success_requests, 'The total number of successful requests to the faucet');
+
+  res.end(`${errors_total}${errors_rpc_timeout}${balance}${total_requests}${successful_requests}`);
```

Comment:
I think joined array looks more reasonable here
```
[
  getMetrics('errors_total', 'counter', errorCounter.total(), 'The total amount of errors logged on the faucet backend'),
  ...
].join('')

Reply:
The question should be whether it works or not. We can delay this PR and make it perfect but it leaves us without metrics in the meantime. Or we start with a working version and you can refactor it if you wish ?
I do agree with you though. The change is simple, it just needs to be tested again then.